src/Common/WorkloadSetVersion.cs

```diff
@@ -8,9 +8,20 @@ namespace Microsoft.DotNet.Workloads.Workload
 {
     static class WorkloadSetVersion
     {
+        public static bool IsWorkloadSetPackageVersion(string workloadSetVersion)
+        {
+
+            string[] sections = workloadSetVersion.Split(['-', '+'], 2);
```

Comment:
The first few lines here are duplicated from below, but they can't really be de-duped, as they are just defining variables. The variables would need to be defined either way.